src/Symfony/Component/Lock/Store/RedisStore.php

```diff
@@ -55,17 +61,82 @@ public function __construct($redisClient, float $initialTtl = 300.0)
     public function save(Key $key)
     {
         $script = '
-            if redis.call("GET", KEYS[1]) == ARGV[1] then
-                return redis.call("PEXPIRE", KEYS[1], ARGV[2])
-            elseif redis.call("SET", KEYS[1], ARGV[1], "NX", "PX", ARGV[2]) then
-                return 1
-            else
-                return 0
+            local key = KEYS[1]
+            local uniqueToken = ARGV[2]
+            local ttl = tonumber(ARGV[3])
+
+            -- asserts the KEY is compatible with current version (old Symfony <5.2 BC)
```

Comment:
This check provide backward compatibility:
- when an old version of symfony put a lock, the new version won't be able to put a lock.

But the opposite is not (always) true:
- with predis: when a new version of symfony put a lock, the old version will fail with an ServerException
- with phpredis it's ok: when an new version of symfony put a lock, the old version won't be able to put a lock.

Both situation are safe (user won't be able to acquire the lock), but the complexity of the code to throw a proper "LockConfictException" instead of "predis\ServerException" don't worth it IMHO